catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsOptionsTest.java

```diff
@@ -0,0 +1,376 @@
+        ArrayList<Result> mockResultList = new ArrayList<>();
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
+                queryResponseMock);
+        when(queryResponseMock.getResults()).thenReturn(mockResultList);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getLocalResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesUnsupportedQueryException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                UnsupportedQueryException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getLocalResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesFederationException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                FederationException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getLocalResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesIllegalArgumentException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                IllegalArgumentException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getLocalResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+
+    }
+
+    private void setUpResourceOperationsAndMocks() throws URISyntaxException {
+        SourceOperations sourceOperations = new SourceOperations(frameworkProperties);
+        OperationsSecuritySupport opsSecurity = new OperationsSecuritySupport();
+        MetacardFactory metacardFactory = new MetacardFactory(mimeTypeToTransformerMapper);
+        OperationsMetacardSupport opsMetacard = new OperationsMetacardSupport(frameworkProperties,
+                metacardFactory);
+
+        QueryOperations queryOperations = new QueryOperations(frameworkProperties,
+                sourceOperations,
+                opsSecurity,
+                opsMetacard);
+
+        resourceOperations = new ResourceOperations(frameworkProperties,
+                queryOperations,
```

Comment:
Should be able to pass in your queryOperationsMock here so you don't need the setQueryOperations function.